llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -101,6 +101,11 @@ static cl::opt<bool> EnableMISchedLoadClustering(
     cl::desc("Enable load clustering in the machine scheduler"),
     cl::init(false));
 
+static cl::opt<bool>
+    EnableSelectOpt("riscv-select-opt", cl::Hidden,
```

Comment:
If no in tree targets use this, should we default to false?

Reply:
We have already disabled it via `enableSelectOptimize()`?

Reply:
But we added a bunch of passes to the pipeline. Does that have compile time impact?

Reply:
Yeah, this point makes sence to me. This pass adds several analysis passes (most of them can be cached), so it may impact compile time.

Reply:
@topperc WDYT?

Reply:
> I think the impact won't be large, since the pass is early out before these analysises actully run when enableSelectOptimize returns false .

The pass manager will run the analysis passes before the runOnFunction in the select optimize pass gets called. Unless those analysis passes do lazy updates and only compute something when they are queried, they will run before the early out.

Reply:
Most of the added passes have been run before, so they may be cached?